src/Nethermind/Nethermind.Network/PeerManager.cs

```diff
@@ -843,39 +846,51 @@ private void StopTimers()
 
         private void CleanupCandidatePeers()
         {
-            if (_peerPool.PeerCount <= _networkConfig.CandidatePeerCountCleanupThreshold)
-            {
-                return;
-            }
+            int peerCount = _peerPool.PeerCount;
 
-            // may further optimize allocations here
-            List<Peer> candidates = _peerPool.NonStaticPeers;
-            int countToRemove = candidates.Count - _networkConfig.MaxCandidatePeerCount;
-            Peer[] failedValidationCandidates = candidates.Where(x => _stats.HasFailedValidation(x.Node))
-                .OrderBy(x => _stats.GetCurrentReputation(x.Node)).ToArray();
-            Peer[] otherCandidates = candidates.Except(failedValidationCandidates).Except(_peerPool.ActivePeers.Values).OrderBy(x => _stats.GetCurrentReputation(x.Node)).ToArray();
-            Peer[] nodesToRemove = failedValidationCandidates.Length <= countToRemove
-                ? failedValidationCandidates
-                : failedValidationCandidates.Take(countToRemove).ToArray();
-            int failedValidationRemovedCount = nodesToRemove.Length;
-            int remainingCount = countToRemove - failedValidationRemovedCount;
-            if (remainingCount > 0)
```

Comment:
Its equivalent to just sorting by: HasFailedValidation desc first and GetCurrentReputation asc second!